### PR TITLE
Add script and docs for preparing releases

### PR DESCRIPTION
**Pull request by AndrewLister-STFC**
I've made a script to ensure tagging is done consistently and added docs around the release process.

Please check the docs make sense and feel free to check the script works (although you may have to undo the 2 commits it will make!)

### CODE REVIEW

File: docs/source/contributors/workflow.rst [hunk 1]
```diff
@@ -94,3 +94,42 @@ the PR that this being backported separately.
 
 If you encounter problems with cherry picking into release branch please
 don't hesitate to speak to an experienced member of the FitBenchmarking team.
+
+
+================
+Create a release
```
Comment (tyronerees) [inline docs/source/contributors/workflow.rst:100]:
```suggestion
Creating a release
```

File: docs/source/contributors/workflow.rst [hunk 2]
```diff
@@ -94,3 +94,42 @@ the PR that this being backported separately.
...
+================
+In order to create a new release for FitBenchmarking, there are a few manual steps.
+These have been streamlined as much as possible.
+
+First checkout the branch to create the release from.
```
Comment (tyronerees) [inline docs/source/contributors/workflow.rst:105]:
```suggestion
First checkout the branch to create the release from.  Releases should only be made from a `release-x-x` branch, not a development branch or master.
```

Comment (tyronerees) [reply docs/source/contributors/workflow.rst:105]: I can't think of a reason why we'd not do this -- can you?  We should probably enforce this in the script.

File: docs/source/contributors/workflow.rst [hunk 3]
```diff
@@ -94,3 +94,42 @@ the PR that this being backported separately.
...
+
+From the root of the repo run the "ci/prep_and_tag_release.sh" script with the new version number.
+The version number will be rejected if it is not of the expected form.
+We expect a "v" followed by the major, minor, and patch numbers,
+and an optional numbered label to mark the type of release.
+
+Possible lables are:
```
Comment (tyronerees) [inline docs/source/contributors/workflow.rst:112]:
```suggestion
Possible labels are:
```

File: docs/source/contributors/workflow.rst [hunk 4]
```diff
@@ -94,3 +94,42 @@ the PR that this being backported separately.
...
+ - -beta (release for testing)
+ - -rc (release candidate)
+ - -dev (release of a development version)
+
```
Comment (tyronerees) [inline docs/source/contributors/workflow.rst:116]: I know we have these three already, but do we need them?  What's the difference between `dev` and `beta`, for example?

File: docs/source/contributors/workflow.rst [hunk 5]
```diff
@@ -94,3 +94,42 @@ the PR that this being backported separately.
...
+and revert the change in a second commit so that the links point back to the latest versions.
+
+These commits will need to be pushed to github.
+
+Finally, you will need to create a release on github.
+This can be done by navigating to the releases page, selecting new release
+and typing in the tag that was given to the release
+(it should tell you the tag exists at this point!).
+
+E.g For a first beta version of release 0.1.0. One would run:
```
Comment (tyronerees) [inline docs/source/contributors/workflow.rst:127]:
```suggestion
For example, For a first beta version of release 0.1.0, one would run:
```

**Review by tyronerees — CHANGES_REQUESTED** [review]
I can't think of a reason why a release wouldn't be made from the branch `release-??`, so I think it's probably worth checking for that in the script.  It's very easy to run stuff from a feature branch by accident, which may cause a few headaches.

Otherwise looks good to me -- I've just made a few minor suggestions on the wordings in the docs.

File: docs/source/contributors/workflow.rst (hunk 2 above)
Comment (AndrewLister-STFC) [reply docs/source/contributors/workflow.rst:105]: I agree, might as well check.

File: docs/source/contributors/workflow.rst (hunk 4 above)
Comment (AndrewLister-STFC) [reply docs/source/contributors/workflow.rst:116]: I was just listing what we already had... Maybe we'd want a `dev` if someone requested a bleeding edge feature and we weren't near a `beta`? We can probably remove this until we need it.

File: docs/source/contributors/workflow.rst [hunk 6]
```diff
@@ -94,3 +94,42 @@ the PR that this being backported separately.
...
+
+From the root of the repo run the "ci/prep_and_tag_release.sh" script with the new version number.
+The version number will be rejected if it is not of the expected form.
+We expect a "v" followed by the major, minor, and patch numbers,
+and an optional numbered label to mark the type of release.
+
+Possible labels are:
+ - -beta (release for testing)
+ - -rc (release candidate)
+ - -dev (release of a development version)
```
Comment (AndrewLister-STFC) [inline docs/source/contributors/workflow.rst:115]:
```suggestion
```

**Review by tyronerees — APPROVED** [review]
all looks good

**Pull request merged**